sql/src/main/java/io/crate/planner/consumer/QueryThenFetchConsumer.java

```diff
@@ -205,6 +210,11 @@ public PlannedAnalyzedRelation visitQueriedTable(QueriedTable table, ConsumerCon
             }
             // HANDLER/MERGE/FETCH related END
 
+            Integer limit = querySpec.limit();
+            if (limit != null && limit + querySpec.offset() > Constants.PAGE_SIZE) {
+                collectNode.downstreamNodes(Collections.singletonList(context.plannerContext().clusterService().localNode().id()));
```

Comment:
hm afaik this should response directly here. imho this must use ExecutionNode.DIRECT_RETURN_DOWNSTREAM_NODE, not?